buku

```diff
@@ -4721,6 +4721,8 @@ def browse(url):
             # Found a GUI browser, suppress browser output
             browse.suppress_browser_output = True
             break
+    if sys.platform == 'win32':  # GUI apps have no terminal IO on Windows
+        browse.suppress_browser_output = False
```

Comment:
Terminal I/O implementation on Windows forces the app to spawn a terminal window for it to actually make any I/O at all; due to this, I/O streams are disabled in pretty much all Windows applications with GUI.

…And manipulating I/O streams somehow breaks console process handle on Windows 11 :sweat_smile:

Reply:
Also, in theory [it should be possible to avoid I/O streams manipulation by setting `browser.redirect_output` & `browser.background` to `True`](https://github.com/python/cpython/issues/66473#issuecomment-1093663254), but a quick practical test showed me that it's ridiculously easy to get a browser instance where this simply doesn't work :-/